internal/api/dto/entitlement.go

```diff
@@ -90,11 +101,54 @@ type EntitlementResponse struct {
 	*entitlement.Entitlement
 	Feature *FeatureResponse `json:"feature,omitempty"`
 	Plan    *PlanResponse    `json:"plan,omitempty"`
+	Addon   *AddonResponse   `json:"addon,omitempty"`
+
+	// TODO: Remove this once we have a proper entitlement entity type
+	PlanID string `json:"plan_id,omitempty"`
 }
 
 // ListEntitlementsResponse represents a paginated list of entitlements
 type ListEntitlementsResponse = types.ListResponse[*EntitlementResponse]
 
+// CreateBulkEntitlementRequest represents the request to create multiple entitlements in bulk
+type CreateBulkEntitlementRequest struct {
+	Items []CreateEntitlementRequest `json:"items" validate:"required,min=1,max=100"`
+}
+
+// CreateBulkEntitlementResponse represents the response for bulk entitlement creation
+type CreateBulkEntitlementResponse struct {
+	Items []*EntitlementResponse `json:"items"`
+}
+
+// Validate validates the bulk entitlement creation request
+func (r *CreateBulkEntitlementRequest) Validate() error {
+	if len(r.Items) == 0 {
+		return ierr.NewError("at least one entitlement is required").
+			WithHint("Please provide at least one entitlement to create").
+			Mark(ierr.ErrValidation)
+	}
+
+	if len(r.Items) > 100 {
+		return ierr.NewError("too many entitlements in bulk request").
+			WithHint("Maximum 100 entitlements allowed per bulk request").
+			Mark(ierr.ErrValidation)
+	}
+
+	// Validate each individual entitlement
+	for i, entitlement := range r.Items {
```

Comment:
when bulk creating, Plan ID (or entity type entity id) will have to be mandatory

can pick in follow up PR